platform/main.roc

```diff
@@ -20,12 +23,74 @@ platform "webserver"
     packages {}
     imports [
         Task.{ Task },
+        Stderr.{ line },
     ]
-    provides [mainForHost]
+    provides [forHost]
 
 import InternalTask
 import Http
 import InternalHttp
 
-mainForHost : InternalHttp.RequestToAndFromHost -> InternalTask.Task InternalHttp.ResponseToHost []
-mainForHost = \request -> request |> InternalHttp.fromHostRequest |> main
+ForHost : {
+    init : Task (Box Model) I32,
+    respond : InternalHttp.RequestToAndFromHost, Box Model -> InternalTask.Task InternalHttp.ResponseToHost [],
+}
+
+forHost : ForHost
+forHost = { init, respond }
+
+init : Task (Box Model) I32
+init =
+    Task.attempt server.init \res ->
+        when res is
+            Ok model -> Task.ok (Box.box model)
+            Err (Exit code str) ->
+                if Str.isEmpty str then
+                    Task.err code
+                else
+                    line str
+                    |> Task.onErr \_ -> Task.err code
+                    |> Task.await \{} -> Task.err code
+
+            Err err ->
+                line
+                    """
+                    Program exited with error:
+                        $(Inspect.toStr err)
+
+                    Tip: If you do not want to exit on this error, use `Task.mapErr` to handle the error.
+                    Docs for `Task.mapErr`: <https://www.roc-lang.org/packages/basic-cli/Task#mapErr>
+                    """
+                |> Task.onErr \_ -> Task.err 1
+                |> Task.await \_ -> Task.err 1
+
+respond : InternalHttp.RequestToAndFromHost, Box Model -> InternalTask.Task InternalHttp.ResponseToHost []
+respond = \request, boxedModel ->
+    when server.respond (InternalHttp.fromHostRequest request) (Box.unbox boxedModel) |> Task.result! is
```

Comment:
For reference the error handling here was inspired from [this zulip discussion](https://roc.zulipchat.com/#narrow/stream/383402-API-Design/topic/Default.20error.20handling.20in.20basic-webserver/near/436375969).

Basically any unhandled Err's returned to the platform will be logged to stderr, and a http 500 error returned. 

This will be a nicer developer experience, particularly for quick demos or beginners who may not want to explicitly handle every error.